integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java

```diff
@@ -0,0 +1,243 @@
+        } catch (AxisFault e) {
+            Assert.assertTrue(e.getMessage().contains("Couldn't get the Synapse Configuration to get the Template"),
+                    "Expected exception not thrown for invalid template");
+        }
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test dynamic template creation service",
+          priority = 10)
+    public void testCreateDynamicTemplate() throws Exception {
+        int prevCount = templateAdmin.getDynamicTemplateCount();
+        templateAdmin.addDynamicSequenceTemplate(KEY,AXIOMUtil.stringToOM(template1));
+        int latestCount = templateAdmin.getDynamicTemplateCount();
+        Assert.assertTrue(prevCount < latestCount, "Dynamic template not added");
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test dynamic template list service",
+          priority = 11)
+    public void testGetDynamicTemplate() throws Exception {
+        OMElement elm = templateAdmin.getDynamicTemplate(KEY);
+        Assert.assertNotNull(elm, "Unable to get dynamic template");
+        Assert.assertTrue(elm.toString().contains(templateName), "Requested template name is invalid");
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test dynamic template creation service",
+          priority = 12)
+    public void testUpdateDynamicTemplate() throws Exception {
+        templateAdmin.updateDynamicTemplate(KEY, AXIOMUtil.stringToOM(template3));
+        templateAdmin.saveDynamicTemplate(KEY, AXIOMUtil.stringToOM(template3));
+        OMElement elm = templateAdmin.getDynamicTemplate(KEY);
+        Assert.assertTrue(elm.toString().contains(dynamicName), "Dynamic template not updated");
+
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test endpoint template creation service",
+          priority = 13)
+    public void testCreateEndpointTemplate() throws Exception {
+        int prevCount = endpointAdmin.getEndpointTemplatesCount();
+        endpointAdmin.addEndpointTemplate(AXIOMUtil.stringToOM(endpointTmpl1));
+        int latestCount = endpointAdmin.getEndpointTemplatesCount();
+        Assert.assertTrue(prevCount < latestCount, "New endpoint template not added");
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test endpoint template list service",
+          priority = 14)
+    public void testListEndpointTemplate() throws Exception {
+        OMElement elm = endpointAdmin.getEndpointTemplate(endpointName);
+        Assert.assertNotNull(elm, "Unable to get requested endpoint template");
+        Assert.assertTrue(elm.toString().contains(endpointName), "New endpoint template not listed");
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test endpoint template update service",
+          priority = 15)
+    public void testUpdateEndpointTemplate() throws Exception {
+        if (!endpointAdmin.hasDuplicateEndpointTemplate(endpointTmpl1)) {
```

Comment:
if endpointAdmin.hasDuplicateEndpointTemplate(endpointTmpl1) returns true, is it a failure or success?
That path is not asserted here.

Reply:
Fixed with d2d9b2c